processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -1474,7 +1474,15 @@ public Rowboat apply(Rowboat lhs, Rowboat rhs)
       Object[] rhsMetrics = rhs.getMetrics();
 
       for (int i = 0; i < metrics.length; ++i) {
-        metrics[i] = metricAggs[i].combine(lhsMetrics[i], rhsMetrics[i]);
+        Object lhsMetric = lhsMetrics[i];
+        Object rhsMetric = rhsMetrics[i];
+        if (lhsMetric == null) {
+          metrics[i] = rhsMetric;
+        } else if (rhsMetric == null) {
+          metrics[i] = lhsMetric;
```

Comment:
I think this looks fine, I'm okay with keeping this approach and ensuring that serializers/aggregators can handle null values.

Alternatively, if you want to write a zero value in this function for null metrics, the AggregatorFactory has a getTypeName() function, you could write 0L or 0.0f based on that in the case where both lhs/rhs are null, but that's probably unnecessary, and it wouldn't cover the complex cases.